pkg/drivers/registry.go

```diff
@@ -0,0 +1,36 @@
+package drivers
+
+import (
+	"context"
+
+	"github.com/k3s-io/kine/pkg/server"
+)
+
+// Constructor is a function that takes a context and a config and returns a leaderElect bool, a server.Backend and an error
+type Constructor func(ctx context.Context, cfg *Config) (leaderElect bool, backend server.Backend, err error)
+
+var driverRegistry map[string]Constructor = map[string]Constructor{}
```

Comment:
revive lint is failing on this:
`var-declaration: should omit type map[string]Constructor from declaration of var driverRegistry; it will be inferred from the right-hand side (revive)`
```suggestion
var driverRegistry = map[string]Constructor{}
```